### PR TITLE
Changes for response notifications

### PR DESCRIPTION
**Pull request by khurshedk**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the `enable_keep_helm` label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CaseService.java [hunk 1]
```diff
@@ -829,10 +832,37 @@ private List<CitizenNotification> getAllCitizenDashboardNotifications(CaseData c
...
+                && "YES".equalsIgnoreCase(caseData.getRespondentResponseNotificationDetails().getAp15NotificationSent()),
+            citizenNotifications,
+            CAN_6B_AP_15
+        );
         return citizenNotifications;
     }
 
+    private void populateCitizenNotifications(boolean caseData, List<CitizenNotification> citizenNotifications,
+                                              String notificationType) {
+        if (caseData) {
```
Comment (yogendraU) [inline src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CaseService.java:859]: what is caseData? please give some meaningful name

File: src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CaseService.java [hunk 2]
```diff
@@ -829,10 +832,37 @@ private List<CitizenNotification> getAllCitizenDashboardNotifications(CaseData c
         } else {
             citizenNotifications.add(CitizenNotification.builder().id(CAN_10_FM5).show(false).build());
         }
-
+        // CAN6,CAN6A,CAN6B
+        populateCitizenNotifications(
```
Comment (yogendraU) [inline src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CaseService.java:836]: Can we discuss on this implementation?

**Review by yogendraU — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CaseService.java (hunk 1 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CaseService.java]: removed as discussed

File: src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CaseService.java (hunk 2 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/services/citizen/CaseService.java]: removed as discussed

File: src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java [hunk 3]
```diff
@@ -186,6 +211,8 @@ public List<DynamicListElement> fetchDocumentDynamicListElements(CaseData caseDa
         }
         //citizen
         if (CollectionUtils.isNotEmpty(caseData.getDocumentManagementDetails().getCitizenQuarantineDocsList())) {
+            log.info("document list from case data ==>> {}", caseData.getDocumentManagementDetails()
```
Comment (yogendraU) [inline src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java:214]: please remove logs

File: src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java [hunk 4]
```diff
@@ -380,14 +500,104 @@ private void sendEmailToCafcassCymru(CaseData caseData, String cafcassCymruEmail
...
+        return bulkPrintService.send(
+            String.valueOf(caseData.getId()),
+            authorisation,
+            LETTER_TYPE,
+            documents,
+            name
+        );
+    }
+
+    public List<Document> getCoverLetters(CaseData caseData, String auth, Address address, String name,
```
Comment (yogendraU) [inline src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java:541]: Please move this piece of code to a generic method that can be used in SOA and Document management

File: src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java [hunk 5]
```diff
@@ -380,14 +500,104 @@ private void sendEmailToCafcassCymru(CaseData caseData, String cafcassCymruEmail
...
+    private GeneratedDocumentInfo fetchApplicantResponseDocument(CaseData caseData, String auth,
+                                                                 Address address, String applicantName,
+                                                                 String respondentName,
+                                                                 boolean isWelsh, String responseDocumentTemplate)
+        throws Exception {
+        Map<String, Object> dataMap = new HashMap<>();
+        dataMap.put("applicantName", null != applicantName ? applicantName : " ");
+        dataMap.put("applicantAddress", address);
+        dataMap.put("id", String.valueOf(caseData.getId()));
+        dataMap.put("respondentName", null != respondentName ? respondentName :
```
Comment (yogendraU) [inline src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java:587]: please move this to constant

File: src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java (hunk 3 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java]: Removed logs

File: src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java (hunk 4 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java]: moved this method to service of application

File: src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java (hunk 5 above)
Comment (khurshedk) [reply src/main/java/uk/gov/hmcts/reform/prl/services/reviewdocument/ReviewDocumentService.java]: moved these to constants as discussed

**Review by channa-patil — APPROVED** [review]

**Pull request merged**